src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -126,11 +132,34 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
   const ethMarketData = useAppSelector(state => selectMarketDataById(state, ethAssetId))
   const foxMarketData = useAppSelector(state => selectMarketDataById(state, foxAssetId))
   const foxAssetPrecision = useAppSelector(state => selectAssetById(state, foxAssetId)).precision
+  const ethAssetPrecision = useAppSelector(state => selectAssetById(state, ethAssetId)).precision
   const lpAssetPrecision = useAppSelector(state =>
     selectAssetById(state, foxEthLpAssetId),
   ).precision
+  const [lpTokenPrice, setLpTokenPrice] = useState<string | null>(null)
   const featureFlags = useAppSelector(selectFeatureFlags)
 
+  useEffect(() => {
+    if (
+      // get price if at least one of lp or farming were on
+      (featureFlags.FoxLP || featureFlags.FoxFarming) &&
+      ethAssetPrecision &&
+      ethMarketData.price &&
+      lpAssetPrecision
+    )
+      getLpTokenPrice(ethAssetPrecision, ethMarketData.price, lpAssetPrecision).then(
```

Comment:
Async IIFE 🙏🏽

Reply:
There is a lot of optimism across this call/promise stack:
- Is this guaranteed to always resolve? This should have a `catch` block
- The consumed `getLpTokenPrice` doesn't catch possible rejections of `getLpTVL` and `uniV2LPContract.totalSupply()`
  - `getLpTVL` doesn't catch rejections of `uniV2LPContract.getReserves()`
  
None of these are ever guaranteed to resolve if e.g we get rugged by Infura for any reason.
This doesn't have to be tackled in this PR, but at the very least this should live in a try/catch.